driver-core/src/main/com/mongodb/internal/connection/ByteBufBsonDocument.java

```diff
@@ -53,38 +53,26 @@ final class ByteBufBsonDocument extends BsonDocument {
 
     private final transient ByteBuf byteBuf;
 
-    static List<ByteBufBsonDocument> createList(final ByteBufferBsonOutput bsonOutput, final int startPosition) {
-        List<ByteBuf> duplicateByteBuffers = bsonOutput.getByteBuffers();
-        CompositeByteBuf outputByteBuf = new CompositeByteBuf(duplicateByteBuffers);
-        outputByteBuf.position(startPosition);
+    // Create a list of ByteBufBsonDocument from a ByteBuf positioned at the start of the first document of an OP_MSG Section
+    // of type Kind 1: Document Sequence
+    // The provided ByteBuf will be positioned at the end of the section upon normal completion of the method
+    static List<ByteBufBsonDocument> createList(final ByteBuf outputByteBuf) {
         List<ByteBufBsonDocument> documents = new ArrayList<>();
-        int curDocumentStartPosition = startPosition;
         while (outputByteBuf.hasRemaining()) {
-            int documentSizeInBytes = outputByteBuf.getInt();
-            ByteBuf slice = outputByteBuf.duplicate();
-            slice.position(curDocumentStartPosition);
-            slice.limit(curDocumentStartPosition + documentSizeInBytes);
-            documents.add(new ByteBufBsonDocument(slice));
-            curDocumentStartPosition += documentSizeInBytes;
-            outputByteBuf.position(outputByteBuf.position() + documentSizeInBytes - 4);
-        }
-        for (ByteBuf byteBuffer : duplicateByteBuffers) {
-            byteBuffer.release();
+            ByteBufBsonDocument curDocument = createOne(outputByteBuf);
```

Comment:
Now this method can simply delegate to `createOne`, removing some code duplication.